File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/implement/PostUpdater.java
package com.eager.questioncloud.core.domain.post.implement;

import com.eager.questioncloud.core.domain.post.model.Post;
import com.eager.questioncloud.core.domain.post.repository.PostRepository;
import com.eager.questioncloud.core.domain.post.vo.PostContent;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PostUpdater {
    private final PostRepository postRepository;

    public void updatePost(Long postId, Long userId, PostContent postContent) {
        Post post = postRepository.findByIdAndWriterId(postId, userId);
        post.updateQuestionBoardContent(postContent);
        postRepository.save(post);
    }
}

File: qc-core/qc-post/qc-post-api/src/main/kotlin/com/eager/questioncloud/post/implement/PostUpdater.kt
package com.eager.questioncloud.post.implement

import com.eager.questioncloud.post.command.ModifyPostCommand
import com.eager.questioncloud.post.domain.PostContent
import com.eager.questioncloud.post.domain.PostFile
import com.eager.questioncloud.post.repository.PostRepository
import org.springframework.stereotype.Component

@Component
class PostUpdater(
    private val postRepository: PostRepository,
) {
    fun modify(command: ModifyPostCommand) {
        val post = postRepository.findByIdAndWriterId(command.postId, command.userId)
        post.updateQuestionBoardContent(
            PostContent.create(
                command.postContent.title,
                command.postContent.content,
                command.postContent.files.map { PostFile(it.fileName, it.url) })
        )
        postRepository.save(post)
    }
}